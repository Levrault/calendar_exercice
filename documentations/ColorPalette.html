<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ColorPalette</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ColorPalette</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ColorPalette<span class="signature">(active, value, onClick)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ColorPalette"><span class="type-signature"></span>new ColorPalette<span class="signature">(active, value, onClick)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Show color palette, change calender selected date's color
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>active</code></td>
            

            <td class="type">
            
                
<span class="param-type">bool</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>value</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onClick</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Palette_ColorPalette.js.html">components/Palette/ColorPalette.js</a>, <a href="components_Palette_ColorPalette.js.html#line8">line 8</a>
    </li></ul></dd>
    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>still need to be a pure componenet ?</li>
        </ul>
    </dd>
    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"> &rarr; {node}</span></h4>
    

    



<div class="description">
    Render
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Palette_ColorPalette.js.html">components/Palette/ColorPalette.js</a>, <a href="components_Palette_ColorPalette.js.html#line21">line 21</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">node</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li><li><a href="Calendar.html">Calendar</a></li><li><a href="ColorPalette.html">ColorPalette</a></li><li><a href="Day.html">Day</a></li><li><a href="EventForm.html">EventForm</a></li><li><a href="EventShow.html">EventShow</a></li><li><a href="Month.html">Month</a></li><li><a href="YearForm.html">YearForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#About">About</a></li><li><a href="global.html#CalendarButton">CalendarButton</a></li><li><a href="global.html#ColorButton">ColorButton</a></li><li><a href="global.html#colorChange">colorChange</a></li><li><a href="global.html#ColorInputField">ColorInputField</a></li><li><a href="global.html#dayReducer">dayReducer</a></li><li><a href="global.html#daySelector">daySelector</a></li><li><a href="global.html#deleteEvent">deleteEvent</a></li><li><a href="global.html#Error">Error</a></li><li><a href="global.html#EventButton">EventButton</a></li><li><a href="global.html#eventCreated">eventCreated</a></li><li><a href="global.html#eventDeleted">eventDeleted</a></li><li><a href="global.html#eventFormDeleteBegin">eventFormDeleteBegin</a></li><li><a href="global.html#eventFormDeleteError">eventFormDeleteError</a></li><li><a href="global.html#eventFormDeleteSuccess">eventFormDeleteSuccess</a></li><li><a href="global.html#eventFormReducer">eventFormReducer</a></li><li><a href="global.html#eventFormReset">eventFormReset</a></li><li><a href="global.html#eventFormSelector">eventFormSelector</a></li><li><a href="global.html#eventFormSubmitBegin">eventFormSubmitBegin</a></li><li><a href="global.html#eventFormSubmitError">eventFormSubmitError</a></li><li><a href="global.html#eventFormSubmitSuccess">eventFormSubmitSuccess</a></li><li><a href="global.html#eventsInit">eventsInit</a></li><li><a href="global.html#eventsReducer">eventsReducer</a></li><li><a href="global.html#eventsSelector">eventsSelector</a></li><li><a href="global.html#fetch">fetch</a></li><li><a href="global.html#fetchAll">fetchAll</a></li><li><a href="global.html#Link">Link</a></li><li><a href="global.html#Loader">Loader</a></li><li><a href="global.html#Main">Main</a></li><li><a href="global.html#mapStateToProps">mapStateToProps</a></li><li><a href="global.html#Message">Message</a></li><li><a href="global.html#Navbar">Navbar</a></li><li><a href="global.html#NavbarRouter">NavbarRouter</a></li><li><a href="global.html#onSelected">onSelected</a></li><li><a href="global.html#PageNotFound">PageNotFound</a></li><li><a href="global.html#post">post</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#Title">Title</a></li><li><a href="global.html#WeekHeader">WeekHeader</a></li><li><a href="global.html#yearFetchAllBegin">yearFetchAllBegin</a></li><li><a href="global.html#yearFetchAllError">yearFetchAllError</a></li><li><a href="global.html#yearFetchAllSuccess">yearFetchAllSuccess</a></li><li><a href="global.html#yearFetchBegin">yearFetchBegin</a></li><li><a href="global.html#yearFetchError">yearFetchError</a></li><li><a href="global.html#yearFetchSuccess">yearFetchSuccess</a></li><li><a href="global.html#yearFormReducer">yearFormReducer</a></li><li><a href="global.html#yearSelector">yearSelector</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Fri Sep 21 2018 13:49:33 GMT+0200 (CEST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>