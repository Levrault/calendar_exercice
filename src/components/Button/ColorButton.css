.button {
  border: 0;
  border-radius: 100%;
  height: resolve($space-m + $space-s);
  outline: none;
  transition: .5s;
  width: resolve($space-m + $space-s);

  &.active,
  &:hover {
    transform: scale(1.2);
  }
}
