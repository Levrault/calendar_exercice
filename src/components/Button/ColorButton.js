import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './ColorButton.css';

/**
 * Color button
 * @param {bool} active
 * @returns {node}
 */
const ColorButton = ({ active, ...rest }) => (
  <button
    type="button"
    className={classnames(styles.container, { [styles.active]: active })}
    style={{ backgroundColor: rest.value }}
    {...rest}
  />
);

ColorButton.propTypes = {
  active: PropTypes.bool
};

export default ColorButton;
