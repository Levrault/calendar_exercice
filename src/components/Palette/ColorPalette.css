.colorPalette {
  @mixin horizontalList;
  flex-wrap: wrap;
  margin-bottom: $space-s;
  max-height: 0;
  opacity: 0;
  overflow-y: hidden;
  transition-duration: 1.5s;
  transition-property: all;
  transition-timing-function: cubic-bezier(0, 1, .5, 1);

  &.active {
    height: resolve($space-m * 7);
    max-height: resolve($space-l * 4);
    opacity: 1;
  }
}

.color {
  display: flex;
  flex-grow: 1;
  justify-content: center;
  margin-top: $space-s;
  width: 25%;
}
