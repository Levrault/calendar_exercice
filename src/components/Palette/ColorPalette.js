import React from 'react';
import ColorButton from '../Button/ColorButton';
import styles from './ColorPalette.css';

// https://material.io/design/color/the-color-system.html#tools-for-picking-colors
const colors = [
  {
    code: '#FF5252',
    name: 'red'
  },
  {
    code: '#FF4081',
    name: 'pink'
  },
  {
    code: '#E040FB',
    name: 'purple'
  },
  {
    code: '#FF6E40',
    name: 'orange'
  },
  {
    code: '#448AFF',
    name: 'blue'
  },
  {
    code: '#64FFDA',
    name: 'teal'
  },
  {
    code: '#B2FF59',
    name: 'green'
  },
  {
    code: '#FFFF00',
    name: 'yellow'
  }, {
    code: '#795548',
    name: 'brown'
  },
  {
    code: '#546E7A',
    name: 'blueGray'
  },
  {
    code: '#0097A7',
    name: 'cyan'
  },
  {
    code: '#FF6F00',
    name: 'amber'
  }
];

const ColorPalette = () => {
  return (
    <ul className={styles.colorPalette}>
      {
        colors.map(({ code, name }) => (
          <li key={name} className={styles.color} >
            <ColorButton type="button" style={{ backgroundColor: code }} />
          </li>
        ))
      }
    </ul>
  );
};

export default ColorPalette;
