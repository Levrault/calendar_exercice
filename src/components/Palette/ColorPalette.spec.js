import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from '../../tools/test/enzyme';
import ColorPalette from './ColorPalette';
import ColorButton from '../Button/ColorButton';
import colors from './colors-constant';

describe('ColorPalette component', () => {
  it('should be hidden when active is false', () => {
    const component = renderer.create(
      <ColorPalette active={false} onClick={() => {}} value={colors[0].code} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when active is true', () => {
    const component = renderer.create(
      <ColorPalette active onClick={() => {}} value={colors[0].code} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick props when a ColorButton is clicked', () => {
    const onClick = jest.fn();
    const component = mount(
      <ColorPalette active onClick={onClick} value={colors[0].code} />
    );
    component.find(ColorButton).first().simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should create has many colorButton has there colors in colors-constant', () => {
    const component = mount(
      <ColorPalette active onClick={() => {}} value={colors[0].code} />
    );
    const colorsBtn = component.find(ColorButton);
    expect(colorsBtn.length).toBe(colors.length);
  });
});
