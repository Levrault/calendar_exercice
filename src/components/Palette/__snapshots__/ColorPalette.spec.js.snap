// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPalette component should be hidden when active is false 1`] = `
<ul
  className=""
>
  <li>
    <button
      className="undefined"
      style={
        Object {
          "backgroundColor": "#FF5252",
        }
      }
      type="button"
      value="#FF5252"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#FF4081",
        }
      }
      type="button"
      value="#FF4081"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#E040FB",
        }
      }
      type="button"
      value="#E040FB"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#FF6E40",
        }
      }
      type="button"
      value="#FF6E40"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#448AFF",
        }
      }
      type="button"
      value="#448AFF"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#64FFDA",
        }
      }
      type="button"
      value="#64FFDA"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#B2FF59",
        }
      }
      type="button"
      value="#B2FF59"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#CDDC39",
        }
      }
      type="button"
      value="#CDDC39"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#795548",
        }
      }
      type="button"
      value="#795548"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#546E7A",
        }
      }
      type="button"
      value="#546E7A"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#0097A7",
        }
      }
      type="button"
      value="#0097A7"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#EEEEEE",
        }
      }
      type="button"
      value="#EEEEEE"
    />
  </li>
</ul>
`;

exports[`ColorPalette component should render correctly when active is true 1`] = `
<ul
  className="undefined"
>
  <li>
    <button
      className="undefined"
      style={
        Object {
          "backgroundColor": "#FF5252",
        }
      }
      type="button"
      value="#FF5252"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#FF4081",
        }
      }
      type="button"
      value="#FF4081"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#E040FB",
        }
      }
      type="button"
      value="#E040FB"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#FF6E40",
        }
      }
      type="button"
      value="#FF6E40"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#448AFF",
        }
      }
      type="button"
      value="#448AFF"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#64FFDA",
        }
      }
      type="button"
      value="#64FFDA"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#B2FF59",
        }
      }
      type="button"
      value="#B2FF59"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#CDDC39",
        }
      }
      type="button"
      value="#CDDC39"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#795548",
        }
      }
      type="button"
      value="#795548"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#546E7A",
        }
      }
      type="button"
      value="#546E7A"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#0097A7",
        }
      }
      type="button"
      value="#0097A7"
    />
  </li>
  <li>
    <button
      className=""
      style={
        Object {
          "backgroundColor": "#EEEEEE",
        }
      }
      type="button"
      value="#EEEEEE"
    />
  </li>
</ul>
`;
